### PR TITLE
test(tools): add unit tests for Bitbucket tools

### LINKED ISSUE #992: Add unit tests for Bitbucket tools (Commits, FileContents, SearchCode)
## Summary

Three Bitbucket tools have no unit tests: `BitbucketCommitsTool`, `BitbucketFileContentsTool`, and `BitbucketSearchCodeTool`. Coverage needs to be added under `tests/tools/`.

## Context

All three tools live under `app/tools/Bitbucket*Tool/` and share helpers from `BitbucketSearchCodeTool` (`_bb_available`, `_bb_creds`, `_resolve_config`). An integration test at `tests/integrations/test_bitbucket.py` already covers the config/validation layer — this issue is about tool-level unit tests.

## What to implement

Create three files:
- `tests/tools/test_bitbucket_commits_tool.py`
- `tests/tools/test_bitbucket_file_contents_tool.py`
- `tests/tools/test_bitbucket_search_code_tool.py`

Each should follow the pattern in `tests/tools/test_grafana_logs_tool.py` and cover:

1. **Contract test** — inherit `BaseToolContract` from `tests/tools/conftest.py`, implement `get_tool_under_test()`
2. **`is_available`** — `True` when `sources["bitbucket"]` has workspace/username/app_password + required field; `False` when creds are missing or key is absent
3. **`extract_params`** — assert the returned dict maps `repo_slug`, `workspace`, `username`, `app_password` correctly
4. **`run`** — patch the underlying `app.integrations.bitbucket` function (e.g., `list_commits`, `get_file_contents`, `search_code`) with `MagicMock` and assert expected output shape

## Acceptance criteria

- [ ] All three test files exist under `tests/tools/`
- [ ] Each covers `is_available`, `extract_params`, and `run` (happy path + missing-creds path)
- [ ] `make test-cov` passes with zero new failures
- [ ] `make lint` passes

## References

- Pattern: `tests/tools/test_grafana_logs_tool.py`
- Tool sources: `app/tools/BitbucketCommitsTool/`, `app/tools/BitbucketFileContentsTool/`, `app/tools/BitbucketSearchCodeTool/`
- Shared helpers: `app/tools/BitbucketSearchCodeTool/__init__.py`
- Integration config: `app/integrations/bitbucket.py`

### PR DESCRIPTION
**Pull request by 7vignesh**
Fixes #992

Added unit tests for the Bitbucket tool entrypoints:
- `BitbucketCommitsTool`
- `BitbucketFileContentsTool`
- `BitbucketSearchCodeTool`

This PR covers:
- tool contract behavior
- `is_available` checks for verified Bitbucket credentials
- `extract_params` mapping for repo, path, query, and credentials
- `run` happy-path behavior
- missing-credentials behavior

I also updated the shared Bitbucket availability helper so the tools require verified Bitbucket credentials, not just a present Bitbucket source.

### Demo/Screenshot for feature changes and bug fixes -
Validation output:
- `python -m pytest tests/tools/test_bitbucket_commits_tool.py tests/tools/test_bitbucket_file_contents_tool.py tests/tools/test_bitbucket_search_code_tool.py -q`
- `python -m ruff check app/ tests/`
- `python -m ruff format --check app/ tests/`
- `python -m mypy app/`

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
Added focused unit tests for each Bitbucket tool using the existing `BaseToolContract` pattern. The tests validate the tool metadata contract, source availability, parameter extraction, happy-path execution, and unavailable-path behavior. I used direct function patching against the Bitbucket integration helpers so the tests stay deterministic and do not make network calls.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] I can explain the purpose of every function, class, and logic block I added
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by 7vignesh** [comment]
@yashksaini-coder 
addressed the bot reviews in the second commit. All checks are passing now.

File: tests/tools/test_bitbucket_commits_tool.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        ),
+        ({}, False),
+    ],
+)
+def test_is_available_requires_repo_and_credentials(sources: dict, expected: bool) -> None:
+    rt = _registered_tool()
+    assert rt.is_available(sources) is expected
+
+
+def test_extract_params_maps_fields() -> None:
```
Comment (yashksaini-coder) [inline tests/tools/test_bitbucket_commits_tool.py:64]: This greptile thread is still open. The parametrize list is missing the case where all four credentials are present but `repo_slug` is absent.

`_list_bitbucket_commits_available` checks `_bb_available(sources) and bb.get('repo_slug', bb.get('repo'))` — so a source dict with verified credentials but no `repo_slug` key should return `False`. That path has no test coverage.

Add one more parametrize case:
```python
(
    {
        "bitbucket": {
            "connection_verified": True,
            "workspace": "acme",
            "username": "bb-user",
            "app_password": "bb-pass",
            # repo_slug intentionally absent
        }
    },
    False,
),
```
This catches regressions where someone accidentally removes the `repo_slug` guard from `is_available`.

File: tests/tools/test_bitbucket_file_contents_tool.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        ),
+        ({"bitbucket": {"connection_verified": True, "workspace": "acme"}}, False),
+        ({"bitbucket": {"connection_verified": True, "repo_slug": "backend-service"}}, False),
+        ({"bitbucket": {"connection_verified": True, "path": "src/main.py"}}, False),
+        ({}, False),
+    ],
+)
+def test_is_available_requires_file_and_credentials(sources: dict, expected: bool) -> None:
+    rt = _registered_tool()
+    assert rt.is_available(sources) is expected
```
Comment (yashksaini-coder) [inline tests/tools/test_bitbucket_file_contents_tool.py:82]: Same issue as the commits tool. `_get_bitbucket_file_contents_available` requires credentials AND both `repo_slug` AND `path`. The test matrix has the case where credentials are present but `path` is absent (sources2), but it's missing the symmetric case: credentials present, `repo_slug` present, `path` absent.

Add:
```python
(
    {
        "bitbucket": {
            "connection_verified": True,
            "workspace": "acme",
            "username": "bb-user",
            "app_password": "bb-pass",
            "repo_slug": "backend-service",
            # path intentionally absent
        }
    },
    False,
),
```
This ensures that if the `bb.get('path')` guard is ever accidentally removed from `is_available`, a test breaks.

**Review by yashksaini-coder — CHANGES_REQUESTED** [review]
Good work overall — all 46 tests pass locally, ruff/mypy/CI all clean, and the env-based flakiness in the missing-credentials tests is correctly fixed by patching `app.tools.BitbucketSearchCodeTool.bitbucket_config_from_env` (which is right, since `_resolve_config` lives there and is shared by all three tools).

Two greptile threads are still open and need small additions before this merges.

**Comment by 7vignesh** [comment]
Hi @yashksaini-coder, addressed both remaining Greptile threads in the latest commit dbccb1a

**Comment by muddlebee** [comment]
@7vignesh fix conflicts pls

**Comment by 7vignesh** [comment]
> @7vignesh fix conflicts pls

@muddlebee fixed
